specs/WebMessageObjects.md

```diff
@@ -157,11 +232,140 @@ interface ICoreWebView2WebMessageReceivedEventArgs2 : ICoreWebView2WebMessageRec
 }
+
+[v1_enum]
+typedef enum COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND {
+  /// FileSystemHandle is for a file (i.e. FileSystemFileHandle)
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_FILE,
+  /// FileSystemHandle is for a directory (i.e. FileSystemDirectoryHandle)
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_DIRECTORY
+} COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION {
+  /// Read-only permission for FileSystemHandle
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READONLY,
+  /// Read and write permissions for FileSystemHandle
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READWRITE
+} COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION;
+
+[uuid(0ecf4d7d-bbf6-4320-930e-82ff6cf2d8dc), object, pointer_default(unique)]
+interface ICoreWebView2FileSystemHandle : IUnknown {
+  /// The Kind of the FileSystemHandle. It can either be a file or a directory.
+  [propget] HRESULT Kind([out, retval] COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND* kind);
+
+  /// The path to the FileSystemHandle.
+  [propget] HRESULT Path([out, retval] LPWSTR* path);
+
+  /// The permissions granted to the FileSystemHandle.
+  [propget] HRESULT Permission([out, retval] COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION* kind);
+}
+
+// Generic container of COM objects to pass to WebView2
+[uuid(d2fba648-109f-401c-b701-44fd3ddfb440), object, pointer_default(unique)]
+interface ICoreWebView2ObjectCollection : ICoreWebView2ObjectCollectionView {
+  /// Removes the object at the specified index.
+  HRESULT RemoveValueAtIndex([in] UINT32 index);
+
+  /// Inserts the object at the specified index.
+  HRESULT InsertValueAtIndex(
+      [in] UINT32 index,
+      [in] IUnknown* value);
+}
+
+[uuid(afae6f48-60d6-4b95-8b81-6f60d9c70f0b), object, pointer_default(unique)]
+interface ICoreWebView2Environment14 : ICoreWebView2Environment13 {
+  /// Create a ICoreWebView2FileSystemHandle from a path. If the `kind` is
+  /// COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_FILE, representation of a
+  /// FileSystemFileHandle will be created, so the `path` must be a path to the
+  /// file or the parent directory of the path must exist. If the `kind` is
+  /// COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_DIRECTORY, representation of a
+  /// FileSystemDirectoryHandle will be created and the path must point to a
+  /// directory or its parent directory must exist.
+  /// In either case, the app must have the same permissions to the path it
+  /// wishes to give the web content to read/write the file/directory. Note that
+  /// these checks will be done when this handle is accessed from web content
+  /// and will cause JS exceptions if they fail.
+  HRESULT CreateWebFileSystemHandle(
+      [in] LPCWSTR path,
+      [in] COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION permission,
+      [in] COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND kind,
+      [out, retval] ICoreWebView2FileSystemHandle** fileSystemHandle);
+
+  /// Create a ICoreWebView2File from a file path.

```

Comment:
Need more documentation here. What happens if the path is invalid, doesn't exist, doesn't point to a file? Does the object keep open a handle to the file? What happens if the file is deleted, moved, is no longer a file after creating the ICoreWebView2File and so on?